contracts/cw-proposal-single/src/contract.rs

```diff
@@ -846,15 +850,54 @@ pub fn reply(deps: DepsMut, env: Env, msg: Reply) -> Result<Response, ContractEr
                 }
                 None => Err(ContractError::NoSuchProposal { id: proposal_id }),
             })?;
-            Ok(Response::new().add_attribute("proposal execution failed", proposal_id.to_string()))
+
+            Ok(Response::new().add_attribute("proposal_execution_failed", proposal_id.to_string()))
         }
         TaggedReplyId::FailedProposalHook(idx) => {
-            let addr = PROPOSAL_HOOKS.remove_hook_by_index(deps.storage, idx)?;
-            Ok(Response::new().add_attribute("removed proposal hook", format!("{addr}:{idx}")))
+            // When firing off proposal hooks indexes `[0,
+            // hook_count)` are used to represent hooks that have been
+            // fired via the regular PROPOSAL_HOOKS method. Index
+            // `hook_count` is used to represent the pre-propose
+            // module.
+            let count = PROPOSAL_HOOKS.hook_count(deps.storage)?;
+            let addr = if idx == count as u64 {
+                match CREATION_POLICY.load(deps.storage)? {
+                    ProposalCreationPolicy::Anyone {} => {
+                        // Something is off if we're getting this
+                        // reply and we don't have a pre-propose
+                        // module installed. This should be
+                        // unreachable.
+                        return Err(ContractError::InvalidHookIndex { idx });
+                    }
+                    ProposalCreationPolicy::Module { addr } => {
+                        // If we are here, our pre-propose module has
+                        // errored while receiving a proposal
+                        // hook. Rest in peace pre-propose module.
+                        CREATION_POLICY.save(deps.storage, &ProposalCreationPolicy::Anyone {})?;
```

Comment:
would we be able to send deposit funds to the dao upon failure and removal of this module?

Reply:
I like still having withdraw as a stopgap, but also was thinking about avoiding steps the dao has to take to handle a software failure